subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -84,12 +84,17 @@ static int wifi_scan(uint32_t mgmt_request, struct net_if *iface,
 	const struct device *dev = net_if_get_device(iface);
 	struct net_wifi_mgmt_offload *off_api =
 		(struct net_wifi_mgmt_offload *) dev->api;
+	struct wifi_scan_params *params = data;
 
 	if (off_api == NULL || off_api->scan == NULL) {
 		return -ENOTSUP;
 	}
 
-	return off_api->scan(dev, scan_result_cb);
+#ifdef CONFIG_WIFI_MGMT_FORCED_PASSIVE_SCAN
+	params->scan_type = WIFI_SCAN_TYPE_PASSIVE;
```

Comment:
We also need a NULL check here, we could either throw an error or as a backward compatible step ignore `params`